.github/workflows/ci.yml

```diff
@@ -0,0 +1,78 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - '**'
+    tags:
+      - 'v*'
+  pull_request:
+  workflow_dispatch:
+
+env:
+  GRADLE_OPTS: >-
+    -Dorg.gradle.project.eu.xenit.cloudsmith.username=${{ secrets.CLOUDSMITH_USER }}
+    -Dorg.gradle.project.eu.xenit.cloudsmith.password=${{ secrets.CLOUDSMITH_APIKEY }}
+    -Dorg.gradle.project.org.alfresco.maven.nexus.username=${{ secrets.ALFRESCO_NEXUS_USERNAME }}
+    -Dorg.gradle.project.org.alfresco.maven.nexus.password=${{ secrets.ALFRESCO_NEXUS_PASSWORD }}
+
+jobs:
+  build-Test-Publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: 11
+          distribution: temurin
+      - name: Login to Xenit Harbor docker registry
+        uses: docker/login-action@v2
+        with:
+          registry: hub.xenit.eu
+          username: ${{ secrets.XENIT_DOCKER_REGISTRY_USERNAME }}
+          password: ${{ secrets.XENIT_DOCKER_REGISTRY_PASSWORD }}
+      - name: Generate build number
+        id: buildNmbr-gen
+        run: echo "buildnumber="$(date +%Y%m%d%H%M) >> $GITHUB_OUTPUT
+      - name: Unit tests
+        uses: gradle/gradle-build-action@v2.3.0
+        with:
+          arguments: :c4a-impl:test -PBuildNumber=${{ steps.buildNmbr-gen.outputs.buildnumber }} --info
+      - name: Integration tests 6x
+        uses: gradle/gradle-build-action@v2.3.0
+        with:
+          arguments: :c4a-test:test-6x:integrationTest -PBuildNumber=${{ steps.buildNmbr-gen.outputs.buildnumber }} --info
+      - name: Integration tests 7x
+        uses: gradle/gradle-build-action@v2.3.0
+        with:
+          arguments: :c4a-test:test-7x:integrationTest -PBuildNumber=${{ steps.buildNmbr-gen.outputs.buildnumber }} --info
+      - name: Publish 6x
+        if: ${{ startsWith(github.ref, 'refs/heads/master') || startsWith(github.ref, 'refs/heads/release') || startsWith(github.ref, 'refs/tags/v') }}
+        uses: gradle/gradle-build-action@v2.3.0
+        env:
+          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.MAVEN_CENTRAL_GPG_KEY }}
+          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.MAVEN_CENTRAL_GPG_PASSWORD }}
+          ORG_GRADLE_PROJECT_sonatype_username: ${{ secrets.MAVEN_CENTRAL_USERNAME }}
+          ORG_GRADLE_PROJECT_sonatype_password: ${{ secrets.MAVEN_CENTRAL_PASSWORD }}
+        with:
+          arguments: >-
+            :c4a-impl:care4alf6x:publishMavenJavaPublicationToSonaTypeRepository 
+            -PBuildNumber=${{ steps.buildNmbr-gen.outputs.buildnumber }} 
+            -PsigningKeyId=CDE3528F
```

Comment:
This automatically signs the build with our private PGP key?

Reply:
yes, the keyring and its pwd are in the secrets, the key identifier needs to be supplied. We could move it to a secret too, but other projects dont, and the protection should be on the keyring level anyways

Reply:
Agreed, it's meant to be public so no sense hiding it